tsweb/tsweb.go

```diff
@@ -253,11 +253,19 @@ type HandlerOptions struct {
 	// is intended to be used to present pretty error pages if
 	// the user agent is determined to be a browser.
 	OnError ErrorHandlerFunc
+
+	// MetricsCallback is called when ServeHTTP is finished and gets
+	// useful data that the implementor can use for metrics.
+	MetricsCallback MetricsCallbackFunc
```

Comment:
I'd call this OnCompletion since you can probably do other things in a callback, not just export metrics :)